app/scripts/metamask-controller.js

```diff
@@ -545,10 +545,18 @@ export default class MetamaskController extends EventEmitter {
         this.networkController.getProviderAndBlockTracker().provider,
       // NOTE: This option is inaccurately named; it should be called
       // onNetworkDidChange
-      onNetworkStateChange: networkControllerMessenger.subscribe.bind(
-        networkControllerMessenger,
-        'NetworkController:networkDidChange',
-      ),
+      onNetworkStateChange: (eventHandler) => {
+        networkControllerMessenger.subscribe(
+          'NetworkController:networkDidChange',
+          () => {
+            const state = this.networkController.store.getState();
+            if (process.env.IN_TEST) {
+              state.providerConfig.chainId = CHAIN_IDS.MAINNET;
```

Comment:
@Gudahtt I noticed your last commit that added this. In which tests is the chain ID getting set to mainnet?

Reply:
None, but the test builds have always used `1` as the chain ID for test updates. See the `getChainId` method for example. I updated this to follow suit since that's how it worked before. That necessitated updating the state to overwrite the chain ID as well, since some of the tests include network switching and this new version of the gas controller uses this state now (whereas before it didn't).

Reply:
Maybe it would be simpler use the real chain ID instead, and update the e2e test mocks to expect the chain ID to be 1337. Hmm.

Reply:
It was simple to remove this; we're now using the real chain ID instead